### ISSUE TITLE
hadoop-hdfs-namenode/datanode errors

### ISSUE THREAD

**Issue by kutschkem** [khrisrichardson/salt-states#16]
In conjuntion with the cloudera manager states, the hadoop-hdfs-*node states are broken. You defined a requirement on a file that is created by the hadoop-hdfs state, but only included the state when the cluster does not run on cloudera manager. I can't tell whether it is on purpose that you excluded the hadoop-hdfs state (actually I think it is, but have no clue why).

Where I am currently at is that I succesfully added an HDFS service and defined roles, but the API calls don't seem to automatically configure the nodes :-( I am trying to use your hdfs states to configure the roles. I do not know whether this is the right approach, do you have input for me?

**Comment by khrisrichardson** [comment]
Considering that I never managed to get the cloudera execution and state modules into a reasonable shape, I likewise never incorporated them into the hadoop, hbase and zookeeper state files.

If you're willing to help whip the execution and state modules into shape, I'll help reincorporate their usage into the state files. I just wish I could recall what was the operation that the CM API failed to address, which made it difficult to replicate what one could do from the UI. I think it was management of the Cloudera Manager monitors.

**Comment by kutschkem** [comment]
Yes the monitors are no regular service, I think, and the APIs don't seem to have a way to set them up properly. You can do it via the REST API, though, but it doesn't hold your hand well.

How did you go about this, did you add the roles yourself and used salt afterwards to manage configuration? Or did you just ditch Cloudera manager completely?

**Comment by kutschkem** [comment]
Actually, there is an API method for that now that I looked closely: http://cloudera.github.io/cm_api/epydoc/5.4.0/cm_api.endpoints.cms.ClouderaManager-class.html#create_mgmt_service

**Comment by khrisrichardson** [comment]
I ditched Cloudera Manager altogether, since the person supervising the project for which I wrote these deprioritized the need for dynamic management. So I had resorted to using salt for formatting and partitioning, and using the CM UI to complete the rest. Was a sad day.
